src/main/java/javaslang/collection/Queue.java

```diff
@@ -645,13 +645,13 @@ public T head() {
     }
 
     @Override
-    public int indexOf(T element) {
-        final int frontIndex = front.indexOf(element);
+    public int indexOf(T element, int from) {
+        final int frontIndex = front.indexOf(element, from);
         if (frontIndex != -1) {
             return frontIndex;
         } else {
             // we need to reverse because we search the first occurrence
-            final int rearIndex = rear.reverse().indexOf(element);
+            final int rearIndex = rear.reverse().indexOf(element, from - front.length());
```

Comment:
I believe it's all right here... we just continue searching in rear part of queue with start < 0